src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs

```diff
@@ -46,7 +46,7 @@ public void AddGlobalSymbolSuppression(ISymbol symbol, SuppressMessageInfo info)
                 }
                 else
                 {
-                    suppressions = new Dictionary<string, SuppressMessageInfo>() {{ info.Id, info }};
+                    suppressions = new Dictionary<string, SuppressMessageInfo>() { { info.Id, info } };
                     _globalSymbolSuppressions.Add(symbol, suppressions);
```

Comment:
isn't dictionary a bit too heavy if we consider a symbol probably has less than 5 suppression associated with it. also, how the thread-safety ensured?